### PR TITLE
fix(runtimeconfig): treat nil values in config merge as zero maps when merging

### PR DESCRIPTION
**Pull request by chaudum**
**What this PR does**:

This PR fixes an issue when merging config files where one file has a `null` value and the other a zero or non-zero map.

Example:

`config1.yaml`:
```yaml
overrides:
  limits: null
```

`config2.yaml`:
```yaml
overrides:
  limits:
    limit_a: 1
```

**Checklist**
- [x] Tests updated

### CODE REVIEW

File: runtimeconfig/manager.go [hunk 1]
```diff
@@ -268,9 +268,22 @@ func mergeConfigMaps(a, b map[string]interface{}, path string) (_ map[string]int
...
+
 		_, aIsMap := a[k].(map[string]interface{})
 		_, bIsMap := b[k].(map[string]interface{})
+
+		if aHasKey && aVal == nil && bIsMap {
+			aIsMap = true
+			out[k] = make(map[string]interface{})
+		}
+
+		if bHasKey && bVal == nil && aIsMap {
```
Comment (chaudum) [inline runtimeconfig/manager.go:282]: `bHasKey` is always `true` and could be omitted.

File: runtimeconfig/manager_test.go [hunk 2]
```diff
@@ -376,6 +376,38 @@ func TestOverridesManagerMultipleFilesWithEmptyFile(t *testing.T) {
...
+  "123": null
+  "456":
+    limit1: 100
+  "789": null
+`,
+			`overrides:
+  "123":
+    limit2: 100
+  "456": null
+  "789": {}
```
Comment (salvacorts) [inline runtimeconfig/manager_test.go:393]: Can we add a third test where the first override is {} and the second is null?

File: runtimeconfig/manager_test.go [hunk 3]
```diff
@@ -376,6 +376,38 @@ func TestOverridesManagerMultipleFilesWithEmptyFile(t *testing.T) {
...
+		Loader:       testLoadOverrides,
+	}
+
+	overridesManager, err := New(overridesManagerConfig, "overrides", nil, log.NewNopLogger())
+	require.NoError(t, err)
+	err = services.StartAndAwaitRunning(context.Background(), overridesManager)
+	require.NoError(t, err)
+
+	merged := overridesManager.GetConfig()
+	t.Log(merged)
```
Comment (salvacorts) [inline runtimeconfig/manager_test.go]: I think we should check here the expected output

**Review by colega — APPROVED** [review]
Thanks
